.github/scripts/generate_pytorch_test_matrix.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+"""Generates a matrix to be utilized through github actions
+
+Will output a matrix to represent our testing configurations, which is currently
+dictated by just sharding.
+
+"""
+
+import json
+import os
+from typing import List
+
+
+NUM_TEST_SHARDS = int(os.getenv('NUM_TEST_SHARDS', '1'))
+
+def generate_sharding_list() -> List[int]:
+    return list(range(1, NUM_TEST_SHARDS + 1))
```

Comment:
Not sure which one is more readable, but list comprehensions are cool ;)
```suggestion
    return [x + 1 for x in range(NUM_TEST_SHARDS)]
```

Reply:
yes i had considered this--do you think one of them is potentially faster?

Reply:
actually on second thought, the number of shards we'd have would really never strain performance so maybe the readability factor is most important

Reply:
for what it's worth, these both seem equally readable to me (and like you said, performance probably doesn't matter)